addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicBridgeHandler.java

```diff
@@ -0,0 +1,410 @@
+
+	public void registerThingListener(VitotronicThingHandler thingHandler) {
+		if (thingHandler == null) {
+			logger.error("It's not allowed to pass a null ThingHandler.");
+		} else {
+			String thingID = thingHandler.getThing().getUID().getId();
+			if (thingHandlerMap.get(thingID) == null) {
+				thingHandlerMap.put(thingID, thingHandler);
+				logger.trace("register thingHandler for thing: {}", thingID);
+				this.sendSocketData("get "+thingID); // update data after register
+			}
+		}
+
+	}
+
+	public void unregisterThingListener(VitotronicThingHandler thingHandler) {
+		if (thingHandler != null) {
+			String thingID = thingHandler.getThing().getUID().getId();
+			if (thingHandlerMap.remove(thingID) == null) {
+				logger.trace("thingHandler for thing: {} not registered",
+						thingID);
+			} 
+		}
+
+	}
+
+	// Background Runables
+
+	private ScheduledFuture<?> pollingJob;
+
+	private Runnable pollingRunnable = new Runnable() {
+		@Override
+		public void run() {
+			logger.trace("Polling job called");
+			if (!isConnect) {
+				startSocketReciever();
+				try {
+					Thread.sleep(5000); // Wait for connection .
+				} catch (InterruptedException e) {
+				}
+			}
+			if (isConnect) {
+				scanThings();
+				refreshData();
+			}
+			
+		}
+
+	};
+
+	private synchronized void startAutomaticRefresh() {
+		if (pollingJob == null || pollingJob.isCancelled()) {
+			pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0,
+					refreshInterval, TimeUnit.SECONDS);
+		}
+	}
+
+	private void refreshData() {
+		logger.trace("Job: refresh Data...");
+		String thingId;
+		for (VitotronicThingHandler thingHandler : new ArrayList<VitotronicThingHandler>(thingHandlerMap.values())) {
+			thingId = thingHandler.getThing().getUID().getId();
+			if (isConnect) {
+				logger.trace("Get Data for '{}'", thingId);
+				sendSocketData("get " + thingId);
+			}
+
+		}
+	}
+
+	// Methods for ThingHandler
+
+	public void scanThings() {
+		logger.trace("Job: Discover Things...");
+		if (!isDiscover) {
+			sendSocketData("list");
+			isDiscover = true;
+		}
+
+	}
+	
+	public ThingStatus getStatus() {
+		return getThing().getStatus();
+	}
+	
+	public void updateChannel(String thingId, String channelId, String value) {
+		sendSocketData("set "+thingId+":"+ channelId + " "+value);	
+	}
+
+	// internal Methods 
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		// No channels - nothing to do
+		logger.info("Handle Command for channelUID: {} Command: {}",
+				channelUID.toString(), command.getClass().getName());
+	}
+
+	@Override
+	public void initialize() {
+		logger.info("Initializing Vitottronic bridge handler{}",
+				this.toString());
+		VitotronicBindingConfiguration configuration = getConfigAs(VitotronicBindingConfiguration.class);
+		ipAddress = configuration.ipAddress;
+		port = configuration.port;
+		refreshInterval = configuration.refreshInterval;
+
+		isDiscover = false;
+		startAutomaticRefresh();

```

Comment:
You should set the thing status to OFFLINE here (before it is still in INITIALIZING)